backend/testfiles/execution/httpclient.tests

```diff
@@ -25,8 +25,8 @@ HttpClient.basicAuth_v1 "" ":" = { Authorization = "Basic Ojo=" }
 // Tests that try to make requests to the internet
 
 // type errors for bad `headers` are OK
-HttpClient.request "get" "https://darklang.com" [1] Bytes.empty = Error "Expected request headers to be a List of (string * string), but got: 1"
-HttpClient.request "get" "https://darklang.com" [("", "")] Bytes.empty = Error "Empty request header key provided"
+HttpClient.request "get" "https://darklang.com" [1] Bytes.empty = Test.typeError_v0 "Expected request headers to be a List of (string * string), but got: 1"
+HttpClient.request "get" "https://darklang.com" [("", "")] Bytes.empty = Test.typeError_v0 "Empty request header key provided"
```

Comment:
The first test here now looks right, but the second one is now wrong I think - an empty string is still a string, so we're not hitting a "type error there," so such should return a `DResult(Error` instead of an `Error`.

In the implementation, where we gather `reqHeaders`, I think we need to change the implementation a bit - I've added some `//` commentary in-line, but suspect we'll need a refactor here to match the expectations.

```fsharp
let reqHeaders : Result<List<string * string>, string> =
  reqHeaders
  |> List.fold (Ok []) (fun agg item ->
    match agg, item with
    | (Error err, _) -> Error err
    | (Ok pairs, DTuple (DStr k, DStr v, [])) ->
      // TODO: what about whitespace? What else can break?
      if k = "" then
        // This case should result in a DResult - it's a valid string, just "bad" input
        Error "Empty request header key provided"
      else
        Ok((k, v) :: pairs)

    | (_, notAPair) ->
      // This case should result in an Error (not DResult)
      // (this is currently resulting in the "correct" behaviour)
      Error $"Expected request headers to be a List of (string * string), but got: {DvalReprDeveloper.toRepr notAPair}")
      |> Result.map (fun pairs -> List.rev pairs)
```

Reply:
@StachuDotNet thanks for comment! I hope understood you correctly. Please look on my last commit.  Probably my solving is naive, but it works :)

Reply:
I think the implementation is great, not naive :)